api/src/pcapi/routes/backoffice_v3/collective_offers/blueprint.py

```diff
@@ -318,5 +322,116 @@ def batch_reject_collective_offers() -> utils.BackofficeResponse:
         flash("L'un des identifiants sélectionnés est invalide", "danger")
         return redirect(request.referrer or url_for("backoffice_v3_web.collective_offer.list_collective_offers"), 303)
 
-    _batch_validate_or_reject_collective_offers(OfferValidationStatus.REJECTED, form.object_ids_list)
+    _batch_validate_or_reject_collective_offers(offer_mixin.OfferValidationStatus.REJECTED, form.object_ids_list)
     return redirect(request.referrer or url_for("backoffice_v3_web.collective_offer.list_collective_offers"), 303)
+
+
+def _is_collective_offer_price_editable(collective_offer: educational_models.CollectiveOffer) -> bool:
+    # if the offer has no stock it has no price therefore its price cannot be updated
+    if not collective_offer.collectiveStock:
+        return False
+
+    # cannot update an offer's price while the cashflow generation script is running
+    if app.redis_client.exists(finance_conf.REDIS_GENERATE_CASHFLOW_LOCK):  # type: ignore [attr-defined]
+        return False
+
+    # cannot update an offer's stock if it is already a processed pricing
+    pricing_query = db.session.query(finance_models.Pricing.id)
+    pricing_query = pricing_query.join(educational_models.CollectiveBooking, finance_models.Pricing.collectiveBooking)
+    pricing_query = pricing_query.join(
+        educational_models.CollectiveStock, educational_models.CollectiveBooking.collectiveStock
+    )
+    pricing_query = pricing_query.filter(
+        educational_models.CollectiveStock.collectiveOfferId == collective_offer.id,
+        finance_models.Pricing.status.in_(
+            (finance_models.PricingStatus.PROCESSED, finance_models.PricingStatus.INVOICED)
+        ),
+    )
+    if pricing_query.one_or_none():
```

Comment:
Pourquoi ne pas utiliser `.exists()` puisqu'on est dans une fonction qui renvoie ce type d'information True/False ?

Reply:
j'avais deja essayé exists mais il me retourne cette erreur dans ce cas : 
`TypeError: Boolean value of this clause is not defined` pour la ligne: `return not pricing_query.exists()` j'ai pas cherché plus loi, de toute façon on ne remonte qu'une colonne ce qui est equivalent a ce que fait le exists (enfin presque, thechniquement remonter une seule colone genere une requete legerement plus opti)

Reply:
Et `return db.session.query(pricing_query.exists()).scalar()` ?